File: src/isd_check/bag.py
class Bag:
    def __init__(self, kode_barang, nama_barang, jenis_barang, stok):
        self.kode_barang = kode_barang
        self.nama_barang = nama_barang
        self.jenis_barang = jenis_barang
        self.stok = stok

    def __str__(self) -> str:
        return f'Kode: {self.kode_barang}, Nama: {self.nama_barang}, Jenis: {self.jenis_barang}, Sisa stok: {self.stok}'

File: src/ISD/Bag.java
package ISD;

public class Bag implements Comparable<Bag> {
    private String kodeBarang, jenisBarang, namaBarang;
    private int stok;

    public Bag(String kodeBarang, String namaBarang, String jenisBarang, int stok) {
        this.kodeBarang = kodeBarang;
        this.namaBarang = namaBarang;
        this.jenisBarang = jenisBarang;
        this.stok = stok;
    }

    public String getKodeBarang() {
        return kodeBarang;
    }

    public String getJenisBarang() {
        return jenisBarang;
    }

    public String getNamaBarang() {
        return namaBarang;
    }

    public int getStok() {
        return stok;
    }

    @Override
    public String toString() {
        return "Bag{" +
                "kodeBarang='" + kodeBarang + '\'' +
                ", jenisBarang='" + jenisBarang + '\'' +
                ", namaBarang='" + namaBarang + '\'' +
                ", stok=" + stok +
                '}';
    }

    @Override
    public int compareTo(Bag o) {
        if (kodeBarang.equals(o.kodeBarang)) {
            return 0; // Kalo sama return-nya 0
        } else {
            return 1; // Kalo beda
        }
    }
}